sql: update `explain` results (#2580)
pingcap/docs: sql-statements/sql-statement-explain.md

![ExplainableStmt](/media/sqlgram/ExplainableStmt.png)

## `EXPLAIN` output format

Currently, `EXPLAIN` in TiDB outputs 5 columns: `id`, `estRows`, `task`, `access object`, `operator info`. Each operator in the execution plan is described by these attributes, with each row in the `EXPLAIN` output describing an operator. The description of each attribute is as follows:

| Attribute name          | Description |
|:----------------|:----------------------------------------------------------------------------------------------------------|
| id            | The operator ID is the unique identifier of the operator in the entire execution plan. In TiDB 2.1, the ID is formatted to display the tree structure of the operator. Data flows from the child node to the parent node. One and only one parent node for each operator. |

> | estRows       | Number of rows that the operator is expected to output. Data is based on statistical information and the evaluation of the operator's logic execution. `estRows` is called `count` in the earlier versions of TiDB 4.0. |

Reviewer: `基于统计信息以及算子的执行逻辑估算而来` means `it is estimated according to statistical information and the operator's logic`.

> | estRows       | Number of rows that the operator is expected to output. Data is based on statistical information and the evaluation of the operator's logic execution. `estRows` is called `count` in the earlier versions of TiDB 4.0. |

Reviewer: `Data is based on statistical information and the evaluation of the operator's logic execution.` -> `It is estimated according to statistical information and the operator's logic`?

| task          | The type of task the operator belongs to. Currently, the execution plans are divided into two tasks: **root** task, which is executed on tidb-server, and **cop** task, which is performed in parallel on TiKV or TiFlash. The topology of the execution plan at the task level is that a root task followed by many cop tasks. The root task uses the output of cop tasks as input. The cop tasks refer to tasks that TiDB pushes down to TiKV or TiFlash. Each cop task is distributed in the TiKV cluster or the TiFlash cluster, and is executed by multiple processes. |

> | access object | Data item information accessed by the operator. The information includes `table`, `partition`, and `index` (if any). Only operators that directly access the data have access to such information. |

Reviewer: `Only operators that directly access the data have access to such information.` -> `Only operators that directly access the data have such information.`?

Author: All updated in 1c79c71c1c013c0972e6183225b5786c306e9a96

> | operator info | Other information about the operator. `operator info` of each operator is different, you can refer to the following examples. |

Reviewer 2: ```suggestion
| operator info | Other information about the operator. `operator info` of each operator is different. You can refer to the following examples. |
```

* TiDB does not support the `EXPLAIN FORMAT=JSON` as in MySQL.
* TiDB does not currently support `EXPLAIN` for insert statements.

## `EXPLAIN FOR CONNECTION`

> `EXPLAIN FOR CONNECTION` is used to get the execution plan of the last executed query in a connection. The output format is the same as that of `EXPLAIN`. However, the implementation of `EXPLAIN FOR CONNECTION` in TiDB is different from that in MySQL. The following lists their differences apart from the output format:

Reviewer 2: ```suggestion
`EXPLAIN FOR CONNECTION` is used to get the execution plan of the last executed query in a connection. The output format is the same as that of `EXPLAIN`. However, the implementation of `EXPLAIN FOR CONNECTION` in TiDB is different from that in MySQL. Their differences (apart from the output format) are listed as follows:
```

> - MySQL returns the **executing** query plan, while TiDB returns the **last executed** query plan.

Reviewer 2: ```suggestion
- MySQL returns the query plan that is **being executing**, while TiDB returns the **last executed** query plan.
```
